ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+
+import ocflib.misc.validators
+from django import forms
+from django.shortcuts import render
+from jinja2 import Environment
+from jinja2 import PackageLoader
+from ocflib.account.search import user_attrs_ucb
+from ocflib.misc.mail import send_mail
+
+from ocfweb.auth import calnet_required
+from ocfweb.component.forms import wrap_validator
+
+
+JINJA_MAIL_ENV = Environment(loader=PackageLoader('ocfweb', ''))
+
+
+class NewReservationRequest(namedtuple(
+    'NewReservationRequest', [
+        'real_name',
+        'contact_email',
+        'student_group',
+        'reason',
+        'date',
+        'starttime',
+        'endtime',
+    ],
+)):
+    """Request for reservation request.
+    :param real_name:
+    :param contact_email:
+    :param student_group:
+    :param reason:
+    :param date:
+    :param starttime:
+    :param endtime:
+    """
+
+    def to_dict(self):
+        return {
+            field: getattr(self, field)
+            for field in self._fields
+        }
+
+
+class RequestForm(forms.Form):
+
+    error_css_class = 'error'
+    required_css_class = 'required'
+
+    real_name = forms.CharField(
+        label='Full Name',
+        widget=forms.TextInput(attrs={'placeholder': 'Jane Doe'}),
+        min_length=3,
+        max_length=32,
+    )
+
+    contact_email = forms.EmailField(
+        label='Contact e-mail',
+        validators=[wrap_validator(ocflib.misc.validators.valid_email)],
+        widget=forms.EmailInput(attrs={'placeholder': 'janedoe@berkeley.edu'}),
+    )
+
+    verify_contact_email = forms.EmailField(
+        label='Confirm contact e-mail',
+        widget=forms.EmailInput(attrs={'placeholder': 'janedoe@berkeley.edu'}),
+    )
+
+    student_group = forms.CharField(
+        label='Student Group',
+        widget=forms.TextInput(attrs={'placeholder': 'OCF'}),
+        min_length=3,
+        max_length=32,
+    )
+
+    reason = forms.CharField(
+        label='Reason for reservation',
+        widget=forms.TextInput(attrs={'placeholder': ''}),
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (mm/dd/yy)',
+        widget=forms.DateInput(attrs={'placeholder': '02/28/18'}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (xx:xx)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (xx:xx)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_contact_email(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email and verify_contact_email:
+            if email != verify_contact_email:
```

Comment:
Not sure I'm following this logic correctly, but it seems like since this is a non-required field, I could just completely leave the verify email field blank?

Reply:
Tested on supernova: By leaving the field blank, a popup occurs asking users to "please fill out this field."
![image](https://user-images.githubusercontent.com/22068978/36924492-982aa922-1e23-11e8-9210-86f86223bd1a.png)

Reply:
Is that something that might be browser-specific?